server/service/src/document/update_trigger.rs

```diff
@@ -0,0 +1,106 @@
+use std::str::FromStr;
+
+use chrono::NaiveDate;
+use repository::{
+    Document, Gender as NameRowGender, NameRow, NameRowRepository, NameStoreJoinRepository,
+    NameStoreJoinRow, NameType, StorageConnection,
+};
+
+extern crate schemafy_core;
+extern crate serde;
+extern crate serde_json;
+
+use serde::{Deserialize, Serialize};
+use util::uuid::uuid;
+
+use super::document_service::DocumentInsertError;
+
+schemafy::schemafy!("src/document/schemas/patient.json");
+
+/// Callback called when the document has been updated
+pub fn document_updated(
+    con: &StorageConnection,
+    store_id: &str,
+    doc: &Document,
+) -> Result<(), DocumentInsertError> {
+    match doc.r#type.as_str() {
+        "Patient" => {
+            let patient: Patient = serde_json::from_value(doc.data.clone()).map_err(|err| {
+                DocumentInsertError::InternalError(format!("Invalid date of birth format: {}", err))
+            })?;
+            let contact = patient.contact_details.get(0);
+            let date_of_birth = match patient.date_of_birth {
+                Some(date_of_birth) => {
+                    Some(NaiveDate::from_str(&date_of_birth).map_err(|err| {
+                        DocumentInsertError::InternalError(format!(
+                            "Invalid date of birth format: {}",
+                            err
+                        ))
+                    })?)
+                }
+                None => None,
+            };
+            let address = patient.addresses.get(0);
+            let name_repo = NameRowRepository::new(con);
+            let patient_name = |first: &Option<String>, last: &Option<String>| {
+                let mut out = vec![];
+                if let Some(first) = first {
+                    out.push(first.clone());
+                }
+                if let Some(last) = last {
+                    out.push(last.clone());
+                }
+                out.join(",")
+            };
+            let existing_name = name_repo.find_one_by_id(&patient.id)?;
+            name_repo.upsert_one(&NameRow {
+                id: patient.id.clone(),
+                name: patient_name(&patient.first_name, &patient.last_name),
+                code: "program_patient".to_string(),
+                r#type: NameType::Patient,
+                is_customer: true,
+                is_supplier: false,
+                supplying_store_id: Some(store_id.to_string()),
```

Comment:
i think this will be removed ?

Reply:
Yes we don't need it, right? I thought supplying_store_id is needed to associate a patient with a store but thats what the name_store_join is doing already.